### DISCUSSION TITLE
Do not return unproven comments

### DISCUSSION THREAD

**Discussion by icecream17** [official-stockfish/Stockfish#3307] [General]
https://github.com/official-stockfish/Stockfish/blob/77eeea407c9d10cf6dbca6f4e07efbbe836c4fb0/src/search.cpp#L832-L837

I'm a bit confused about this comment, it looks like the opposite thing is happening. 
The code *dis*allows proven wins right???

It returns positive eval, and for example VALUE_MATE > VALUE_KNOWN_WIN... so I don't know what it could possibly be.

**Comment by noobpwnftw** [comment]
This code is to perform pruning, which means it has not yet performed any search to "prove" that any value is in fact truth. For any value > VALUE_KNOWN_WIN, it is bad to assume such unproven results, therefore the above condition does what it says it does and the logic is correct.

**Reply by icecream17** [reply]
Oh I see. Thanks
